compiler/rustc_resolve/src/rustdoc.rs

```diff
@@ -454,7 +455,7 @@ fn parse_links<'md>(doc: &'md str) -> Vec<Box<str>> {
         }
     }
 
-    for (label, refdef) in event_iter.reference_definitions().iter() {
+    for (label, refdef) in event_iter.reference_definitions().iter().sorted_by_key(|x| x.0) {
         if !refids.contains(label) {
```

Comment:
Btw, if you want to, you could additionally harden this by swapping out the `FxHashSet` (the type of `refids`) with `UnordSet` which is just a wrapper around the former that forbids any sort of iteration

Reply:
Sure